File: cs50x/session2020/problemSet1/mariomore/mario.c
/**
|-------------------------------------------------------------------------------
| mario.c
|-------------------------------------------------------------------------------
|
| Author:       Alwin Tareen
| Created:      Jun 30, 2020
| Compilation:  make mario
| Execution:    ./mario
| Check50:      check50 cs50/problems/2020/x/mario/more
| Submit50:     submit50 cs50/problems/2020/x/mario/more
|
| This program creates a pyramid of hash marks.
|
*/

#include <stdio.h>

int main(void)
{
    int num;

    do
    {
        printf("Height: ");
        scanf("%d", &num);
    } while (num < 1 || num > 8);

    printf("Stored: %d\n", num);

    for (int i = 1; i <= num; i++)
    {
        for (int j = num-i; j >= 1; j--)
        {
            printf(" ");
        }

        for (int k = 1; k <= i; k++)
        {
            printf("#");
        }

        printf("  ");

        for (int m = 1; m <= i; m++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: cs50x/session2020/problemSet6/mariomore/mario.py
###
#-------------------------------------------------------------------------------
# mario.py
#-------------------------------------------------------------------------------
#
# Author:       Alwin Tareen
# Created:      Nov 24, 2020
# Execution:    python3 mario.py
# Submit50:     submit50 cs50/problems/2020/x/sentimental/mario/more
#
# This program creates a pyramid of hash marks.
#
##

height = input("Height: ")
while height not in "12345678" or height == "":
    height = input("Height: ")

height = int(height)
offset = height-1
for level in range(1, height+1):
    for space in range(offset, 0, -1):
        print(" ", end="")
    for item in range(level):
        print("#", end="")
    print("  ", end="")
    for item in range(level):
        print("#", end="")
    print()
    offset -= 1

